woocommerce/woocommerce-ios
# [Just In Time Messages] Add build and locale identifiers to JITM request

Closes: #8015


## Description

In Project Salas, we've added the ability to send Just In Time Messages to all Woo Mobile users, initially displayed on the dashboard. A single message definition will be picked up both on iOS and Android. There are also cases when we want to send a message to a smaller subset of our users, or define test JITMs which are not yet ready for production. The `with_query_string` and `without_query_string` JITM rules can help here, but only if we pass useful data in the query.

### JITM Query
To take advantage of these rules, this PR adds the following to the GET `/jetpack/v4/jitm` request, in the `query` part of the `query` object, which is passed to the JITM engine:

- `platform` = `ios`
- `version` = `11.1`
- `build_type` = `developer` for `localDeveloper` and CI builds. N.B. we don't include this parameter on alpha/beta/production builds.
- `device` = `phone` or `pad`

### Locale
There is undocumented support for a `locale` query parameter (top level) on our API. This is not supported in the JITM endpoints at present, but passing it now will allow us to support locale mismatch between the site and phone in future. At present, all JITMs will be localised to the site or user's default language.

`locale=en_US` or other phone locale code. This deliberately excludes any currency or other locale identifier information.

See pdfdoF-1Fe-p2 for full context.

## Testing instructions


Configure Proxyman/Charles to view your network traffic, or use the WormHoly tools in the settings menu to view the requests later.

1. Run the app
2. Observe that a request is made to `/jetpack/v4/jitm`
3. Observe that the request has a `locale` parameter with your phone's locale as the value, at the top level of the request – `1` in screenshot below
4. Observe that the request has a `query` parameter, with a JSON object, with an internal `query` key.
5. Observe that the value for that key has `platform`, `version`, `build_type`, and `device` values as specified above – `2` in screenshot below.

N.B. the values for these parameters, including locale, are not currently used for anything by the backend.

## Screenshots

![CleanShot 2022-11-03 at 14 28 53@2x](https://user-images.githubusercontent.com/2472348/199752803-19e2d759-9a1f-4b5c-a674-7ed4b1e4b0d8.png)


---
